### ISSUE TITLE
RichText: discourage direct getting/setting of value properties

### ISSUE THREAD

**Issue by iseulde** [WordPress/gutenberg#12007]
Previously: #10327.

In the meantime the use of the rich text value has changed: it is mostly internal to `RichText` and no longer sent to editor state or the server. The value leaves `RichText` as HTML. But we do expose it through the format API. Would prefer if we could take some measures to discourage direct getting and setting of the object properties, both to prevent subtle bugs (thinking about altering the formats property), and to future-proof it.

* We could prefix the value with an underscore. Seems like the easiest solution for now.
* We could create a new `RichTextValue` type to pass around and keep the values private. Since we're no longer storing the value anywhere this is now possible. Previously it was stored in the editor state and preferred to be a plain object. A benefit of this is also that methods are easily available to plugin authors. Cc @dmsnell.

Adding to 4.5 because this should be done as soon as possible.

**Comment by aduth** [comment]
What's the difference between something like this and directly setting properties on `attributes`, or accessing / setting properties on a React element in some `withFilters` callback? Seems similar to conversations at #10890 and particularly my broad concerns of ad-hoc enforcement / guard-rails raised in https://github.com/WordPress/gutenberg/pull/10890#issuecomment-434473436.

tl;dr : I don't think we should do it.

**Comment by iseulde** [comment]
Okay, closing. I hope not documenting the value makes it clear enough that all properties are for internal use only.

**Comment by aduth** [comment]
Summarizing a direct message conversation, there are some legitimate arguments in favor of doing _something_. I'm still of the mind that any steps taken here should be made with consideration of _all_ APIs, not RichText specifically. My general leaning is still in keeping things as they are.

- We do have conventions around [Experimental APIs](https://github.com/WordPress/gutenberg/blob/master/docs/reference/coding-guidelines.md#experimental-apis), which seems in concurrence with marking a property as internal
   - These, however, are more relevant to the public interface of a module or function, as opposed to the generated result of said functions
- What obligations do we have to support things which are not publicly documented?
   - [SemVer](https://semver.org/) specifically considers a breaking change _in relation to a public API_.
   - Though in the real world, if it's there, people may use it.
   - Does adding a prefix remit us from the responsibility of maintaining compatibility?
- How much of this could be attributed to a classic fear of labelling something as 1.0, forever perpetuating a 0.x lifecycle to avert committing to an API
